docs/ide-settings.md

```diff
@@ -0,0 +1,152 @@
+
+# ccls Installation
+
+## Install ccls on CentOS
+
+> NOTE: if you don't have the permission for `yum`, please use `sudo yum ...` instead.
+
+```bash
+yum install epel-release
+yum install snapd # On centos8: yum install snapd --nobest 
+systemctl enable --now snapd.socket
+ln -s /var/lib/snapd/snap /snap
+snap install ccls --classic
+```
+
+And then add the command below into your env source file, such as '~/.bashrc' or '~/.bash_profile'
+
+```bash
+export PATH=/var/lib/snapd/snap/bin:$PATH
+```
+
+Now, refresh your environment like this:
+
+```bash
+source ~/.bashrc   # or
+source ~/.bash_profile
+```
+
+## Install ccls on Ubuntu
+
+```bash
+apt-get -y install ccls
+```
+
+> NOTE: If you don't have the permission, please use `sudo` as the command prefix.
+
+## Check the Installation
+
+You can run the command below to check whether the installation was success.
+
+```bash
+ccls --version
+```
+
+# VSCode Configuration
+
+## Remote Plugin
+
+Once the source code has been located in the remote machine, it is easy to setup debugging environment in remote machine. At the same time, the application can be run faster because remote machine is more powerful.  User can easily access the source code on the remote machine even when something is wrong with the network, just wait reload after reconnect the remote server. 
+
+### Installation
+
+Download and install the Remote plugin from the VSCode extension store.
+
+![remote plugin](images/ide-settings-remote-plugin.png)
+
+### Usage
+
+**NOTE**：Make sure the connection between the local machine and the remote machine is fine. 
+After installation the plugin, there is one icon in the left bottom corner of VSCode. 
+
+![remote plugin usage](images/ide-settings-remote-plugin-usage.png)
+
+Press the icon and select `Connect to Host`, or press shortkey `ctrl+shift+p` and select `Remote-SSH:Connect to Host`:
+
+![connec to remote ](images/ide-settings-connect-to-remote-server.png)
+
+Input user@remote_ip in the input frame, VSCode will create one new window, please input password in the new window:
+
+![input password](images/ide-settings-input-password.png)
+
+After input the password, VSCode will connect to the remote server, and it is ready to open the remote machine's file or directory. 
+
+If you want to use the specific port, please choose `Add New SSH Host`, then input ssh command, then choose one configuration file to store the ssh configuration. 
+
+![ssh port](images/ide-settings-use-different-ssh-port.png)
+
+![ssh config file](images/ide-settings-choose-ssh-config.png)
+After that, the configured machines can be found in the `Connect to Host`. 
+
+Password need to be input everytime. If you want to skip this action, please configure SSH security login with credential. 
+
+## C/C++ Plugin
+
+C/C++ plugin can be download and installed in VSCode extension store in the case of simple scenarios:
```

Comment:
it is an optional ? as C/C++ plugin is not compatible with ccls plugin

Reply:
When we write code with vscode, we can use c/c++ pulgin or ccls plugin, so it's an option.

I added our advice above this information.